lib/http/http.c

```diff
@@ -559,8 +560,11 @@ gotheaders(struct http_cookie * H, uint8_t * buf, size_t buflen)
 	 */
 	if ((clen = http_findheader(H->res.headers, H->res.nheaders,
 	    "Content-Length")) != NULL) {
-		/* Parse the value (strtoull skips LWS). */
-		len = strtoull(clen, NULL, 0);
+		/* Parse the value. */
+		if (PARSENUM(&len, clen)) {
+			warn0("Invalid Content-Length: %s", clen);
```

Comment:
Should we use `warnp` here to print ERANGE vs. EINVAL?  I went with a generic "Invalid", rather than "too long", because the latter wouldn't apply to EINVAL.